_test/test_sqw/test_PixelData_operations.m

```diff
@@ -87,31 +95,172 @@ function test_compute_bin_data_correct_output_5_pages_mex_1_thread(obj)
         pix = PixelData(obj.test_sqw_file_path, pg_size);
         [s, e] = pix.compute_bin_data(obj.ref_npix_data);
 
-        assertEqual(s, obj.ref_s_data, '', 4.75e-4);
-        assertEqual(e, obj.ref_e_data, '', 4.75e-4);
+        assertEqual(s, obj.ref_s_data, '', obj.FLOAT_TOLERANCE);
+        assertEqual(e, obj.ref_e_data, '', obj.FLOAT_TOLERANCE);
     end
 
     function test_compute_bin_data_correct_output_file_backed_mex_4_threads(obj)
         obj.config.use_mex = true;
         obj.config.threads = 4;
 
-        pix = PixelData(obj.test_sqw_file_path, obj.page_size);
-        [s, e] = pix.compute_bin_data(obj.ref_npix_data);
+        [s, e] = obj.pix_with_pages.compute_bin_data(obj.ref_npix_data);
 
-        assertEqual(s, obj.ref_s_data, '', 4.75e-4);
-        assertEqual(e, obj.ref_e_data, '', 4.75e-4);
+        assertEqual(s, obj.ref_s_data, '', obj.FLOAT_TOLERANCE);
+        assertEqual(e, obj.ref_e_data, '', obj.FLOAT_TOLERANCE);
     end
 
     function test_compute_bin_data_correct_output_file_backed_mex_off(obj)
         obj.config.use_mex = false;
 
-        pix = PixelData(obj.test_sqw_file_path, obj.page_size);
-        [s, e] = pix.compute_bin_data(obj.ref_npix_data);
+        [s, e] = obj.pix_with_pages.compute_bin_data(obj.ref_npix_data);
+
+        assertEqual(s, obj.ref_s_data, '', obj.FLOAT_TOLERANCE);
+        assertEqual(e, obj.ref_e_data, '', obj.FLOAT_TOLERANCE);
+    end
+
+    function test_do_unary_op_returns_correct_output_with_cosine_gt_1_page(obj)
+        data = rand(9, 50);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        pix = pix.do_unary_op(@cos_single);
+
+        % Loop back through and validate values
+        pix.move_to_first_page();
+        iter = 0;
+        while true
+            start_idx = (iter*npix_in_page) + 1;
+            end_idx = min(start_idx + npix_in_page - 1, pix.num_pixels);
+
+            original_signal = data(obj.SIGNAL_IDX, start_idx:end_idx);
+            original_variance = data(obj.VARIANCE_IDX, start_idx:end_idx);
+
+            expected_data = data;
+            expected_data(obj.SIGNAL_IDX, start_idx:end_idx) = ...
```

Comment:
A comment here explaining where the signal and variance calculations are taken from would make this explicit